R/conditions.R

```diff
@@ -8,7 +8,8 @@
 #' @param details Any additional human readable details
 #' @param subclass Use if you want to further customise the class
 #' @param locations For `warn_lossy_cast()`, an optional vector giving the
-#'   locations where `x` lost information.
+#'   locations where `x` lost information.  If the vector is given and empty,
+#'   no warning is issued.
```

Comment:
Probably needs a `@value` section where it states it's called for its side-effect, and it only warns if there is some loss